src/Data/NodeResolver.php

```diff
@@ -112,22 +277,33 @@ public function resolve_uri( string $uri, $extra_query_vars = '' ) {
 			}
 		}
 
-		$this->wp->query_vars = [];
-		$post_type_query_vars = [];
-
-		if ( isset( $parsed_url['query'] ) && '/' === $parsed_url['path'] ) {
+		if ( isset( $parsed_url['query'] ) && ( empty( $parsed_url['path'] ) || '/' === $parsed_url['path'] ) ) {
 			$uri = $parsed_url['query'];
 		} elseif ( isset( $parsed_url['path'] ) ) {
 			$uri = $parsed_url['path'];
 		}
 
+		/**
+		 * Follows pattern from WP::parse_request()
+		 *
+		 * @see https://github.com/WordPress/wordpress-develop/blob/6.0.2/src/wp-includes/class-wp.php#L135
+		 */
+		global $wp_rewrite;
+
+
+		$this->wp->query_vars = [];
+		$post_type_query_vars = [];
+
+
 		if ( is_array( $extra_query_vars ) ) {
 			$this->wp->query_vars = &$extra_query_vars;
```

Comment:
I _think_ this is a preexisting bug, and it should be `$this->wp->extra_query_vars`, however without good unit tests I am scared to change it in this pr.

See https://github.com/WordPress/wordpress-develop/blob/6.0.2/src/wp-includes/class-wp.php#L155